front/src/views/user/eating/eating-new/eating-new.view.js

```diff
@@ -0,0 +1,171 @@
+  components: {
+    VCol,
+    VRow,
+    VBtn,
+    VIcon,
+    VTextField,
+    VMenu,
+    VDatePicker,
+    MmtTextInput,
+    VTimePicker,
+    VForm,
+    VFileInput
+  },
+  data () {
+    return {
+      menuDatePicker: false,
+      menuTimePicker: false,
+      formValid: false,
+      rules: {
+        maxLengthName: v => v.length <= 200 || this.$t('eatingNew.maxLengthName'),
+        maxLengthNotes: v => v.length <= 2000 || this.$t('eatingNew.maxLengthNotes'),
+        linkValid: v => {
+          if (v) {
+            return RegExp(LINK_REGEX).test(v) || this.$t('eatingNew.isNoLink');
+          } else {
+            return true;
+          }
+        }, // правило на 10 мб
+        fileValid: v => {
+          if (v) {
+            let valid = true;
+            v.forEach(file => {
+              if (file.size > MAX_FILE_SIZE_SCHEDULET_PLACE_TO_EAT) {
+                valid = false;
+              }
+            });
+            return valid ? true : 'File size should be less than 10 MB!';
+          } else {
+            return true;
+          }
+        },
+        fileLength: v => {
+          if (v) {
+            return v.length <= 10 || 'Количество вложений не должно быть больше 10';
+          } else {
+            return true;
+          }
+        }
+      },
+      date: new Date().toISOString().substr(0, 10), // new Date().toLocaleDateString()
+      time: new Date().toTimeString().substr(0, 5),
+      notes: '',
+      link: '',
+      namePlace: '',
+      googlePlaceId: '',
+      lat: 0,
+      lng: 0,
+      files: new FormData(),
+      text: '',
+      snackbar: false,
+      color: ''
+    };
+  },
+  computed: {
+    google: gmapApi
+  },
+  methods: {
+    findPlace (place) {
+      this.googlePlaceId = place.place_id || '';
+      if (place.geometry) {
+        this.lat = place.geometry.location.lat();
+        this.lng = place.geometry.location.lng();
+      }
+    },
+    parseDate (date) {
+      if (!date) {
+        return null;
+      }
+
+      const [month, day, year] = date.split('/');
+      return `${year}-${month.padStart(2, '0')}-${day.padStart(2, '0')}`;
+    },
+    save () {
+      const payload = {
+        dateTime: `${this.date}T${this.time}`,
+        namePlace: this.namePlace,
+        notes: this.notes,
+        link: this.link,
+        googlePlaceId: this.googlePlaceId,
+        lat: this.lat,
+        lng: this.lng
+      };
+      if (this.$route.params.id) {
+        payload.id = Number(this.$route.params.id);
+        this.$store.dispatch('eating/updateEating', payload)
+          .then(() => {
+            this.text = this.$t('eatingNew.successfullyEdited');
+            this.snackbar = true;
+            this.color = 'success';
+            this.$store.dispatch('eating/uploadNewFilesEating', {
+              files: this.files,
+              eatingId: payload.id
+            });
+            setTimeout(() => this.$router.push({ name: 'ScheduleEatingPlace' }), 3000);
```

Comment:
why do you need timeout? might you need nextTick()

Reply:
snackBar doesn't have time to show up

Reply:
it was absolutely not obvious 
the right thing is to make 
```
const SNACK_BAR_TIMEOUT = 4000;
// ...
showSnackbar(color, message) {
this.snackBarColor = color;
  this.snackBarMessage = message;
  this.showSnackBar = true;
  return new Promise(function(resolve, reject) {
    setTimeout(() => resolve(true), SNACK_BAR_TIMEOUT);
  });
}

//..

this.$store.dispatch(...)
  .then(() => {
    ...
    return this.$store.dispatch(...)
  })
  .then(showSnackBar)
  .then(() => this.$router.push(...))
  .catch(e => handleException(e));
```

but actually it is not pretty well to make people to wait page opening because it need to show message
this snackbar shouldn't belong to your view - it should be at common layout as container component (or clever component) and have actions and state in store - something like notification module